AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/Support/MutationEventExtensionsTests.swift

```diff
@@ -199,4 +239,317 @@ class MutationEventExtensionsTest: BaseDataStoreTests {
+            guard case .success = result else {
+                XCTFail("Failed to save metadata")
+                return
+            }
+            updateMutationExpectation1.fulfill()
+        }
+
+        let updateMutationExpectation2 = expectation(description: "save updateMutationEvent2 success")
+        storageAdapter.save(updateMutationEvent2) { result in
+            guard case .success = result else {
+                XCTFail("Failed to save metadata")
+                return
+            }
+            updateMutationExpectation2.fulfill()
+        }
+
+        wait(for: [updateMutationExpectation1, updateMutationExpectation2], timeout: 1)
+
+        let queryBeforeUpdatingVersionExpectation = expectation(description: "update mutation should have version 1")
+        let queryAfterUpdatingVersionExpectation = expectation(description: "update mutation should have version 1")
+        let updatingVersionExpectation = expectation(description: "don't update latest mutation event with response version")
+
+        MutationEvent.pendingMutationEvents(for: post1.id,
+                                            storageAdapter: storageAdapter) { result in
+            switch result {
+            case .failure(let error):
+                XCTFail("Error : \(error)")
+            case .success(let mutationEvents):
+                guard !mutationEvents.isEmpty, let head = mutationEvents.first else {
+                    XCTFail("Failure while updating version")
+                    return
+                }
+                XCTAssertEqual(head.version, 1)
+                XCTAssertEqual(head.mutationType, MutationEvent.MutationType.update.rawValue)
+                queryBeforeUpdatingVersionExpectation.fulfill()
+            }
+        }
+        wait(for: [queryBeforeUpdatingVersionExpectation], timeout: 1)
+
+        MutationEvent.reconcilePendingMutationEventsVersion(mutationEvent: updateMutationEvent1,
+                                                             mutationSync: updateMutationSync,
+                                                             storageAdapter: storageAdapter) { result in
+            switch result {
+            case .failure(let error):
+                XCTFail("Error : \(error)")
+            case .success:
+                updatingVersionExpectation.fulfill()
+            }
+        }
+        wait(for: [updatingVersionExpectation], timeout: 1)
+
+        // query for head of mutation event table for given model id and check if it has the correct version
+        MutationEvent.pendingMutationEvents(for: post1.id,
+                                            storageAdapter: storageAdapter) { result in
+            switch result {
+            case .failure(let error):
+                XCTFail("Error : \(error)")
+            case .success(let mutationEvents):
+                guard !mutationEvents.isEmpty, let updatedEvent = mutationEvents.first else {
```

Comment:
if it wasn't supposed to be updated, `updatedEvent` is confusing, perhaps just `pendingMutationEvent` or `mutationEventHead`